pkg/pxc/service.go

```diff
@@ -212,22 +212,37 @@ func NewServiceProxySQLUnready(cr *api.PerconaXtraDBCluster) *corev1.Service {
 
 func NewServiceProxySQL(cr *api.PerconaXtraDBCluster) *corev1.Service {
 	svcType := corev1.ServiceTypeClusterIP
-	if cr.Spec.ProxySQL != nil && len(cr.Spec.ProxySQL.ServiceType) > 0 {
-		svcType = cr.Spec.ProxySQL.ServiceType
+
+	if cr.Spec.ProxySQL != nil {
+		if cr.CompareVersionWith("1.14.0") >= 0 && len(cr.Spec.ProxySQL.Expose.Type) > 0 {
+			svcType = cr.Spec.ProxySQL.Expose.Type
+		} else if len(cr.Spec.ProxySQL.ServiceType) > 0 {
+			svcType = cr.Spec.ProxySQL.ServiceType
+		}
```

Comment:
```suggestion
		if cr.CompareVersionWith("1.14.0") >= 0 {
			if len(cr.Spec.ProxySQL.Expose.Type) > 0 {
				svcType = cr.Spec.ProxySQL.Expose.Type
			}
		} else {
			if len(cr.Spec.ProxySQL.ServiceType) > 0 {
				svcType = cr.Spec.ProxySQL.ServiceType
			}
		}
```

I think we should not use deprecated value even if the new one is empty.

Reply:
But if the user uses 1.14.0 and has `cr.Spec.ProxySQL.ServiceType` set to something (he shouldn't, but he does, regardless), should we ignore that set for him? Isn't that breaking backward compatibility? Because the old field is just deprecated, but the user can still use it until it is removed.

@egegunes @hors wdyt?

Reply:
The main idea is not add any new fields after operator update to avoid restarts. In this case these new fields will work but will not be added. So, I am ok with it.

Reply:
If we agree on to use deprecated values when the new ones are not set, should we keep this logic consistent in other parts of the code, i.e. here: https://github.com/percona/percona-xtradb-cluster-operator/blob/6f3bd96cc9211dad86dc1a7ec85cd25d3b1ce7be/pkg/apis/pxc/v1/pxc_types.go#L872-L883

Reply:
@pooknull I rechecked all of these and I think we are good.